dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/cdi/OpenTelemetryProducer.java

```diff
@@ -31,39 +33,65 @@ public class OpenTelemetryProducer {
 
     private static final TraceComponent tc = Tr.register(OpenTelemetryProducer.class);
 
-    J2EEName j2EEName = null;
-
-    private J2EEName getJ2EEName() {
-        if (j2EEName == null) {
-            j2EEName = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getJ2EEName();
-        }
-        return j2EEName;
-    }
+    ApplicationMetaData metaData = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getModuleMetaData().getApplicationMetaData();
```

Comment:
1. Field should be private
2. Because this class is a normal scoped bean, we should avoid doing things in the no-op constructor or in field initializers since that code will be called when the proxy is created.
   
   It would be better to initialize this field in a `@PostConstruct` method.

Reply:
Check